### PR TITLE
improve readme and add overview about module backend part

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

**Comment by jreidinger** [comment]
@mvidner you are right. lets improve it. BTW is possible to have svg with clickable links?

File: README.md [hunk 1]
```diff
@@ -5,4 +5,43 @@
...
+
+### High Level Overview
+
+The bootloader module consists of two more or less separated components.
+The **UI** including dialogs, widgets and similar, and
+the **backend** that is responsible for reading, writing, proposing.
+
+### Backend
+
+![overview picture](doc/bootloader_backend.svg)
```
Comment (mvidner) [inline README.md]: SVG on GitHub is tricky: https://github.com/github/markup/issues/556
Maybe you want to add a note:

> To get to the clickable links, right-click the diagram and choose "Display Image" (Firefox) or "Open Image in New Tab" (Chromium).

Comment (mvidner) [reply README.md]: Also note a trick with raw.github.com/...foo.svg?sanitize=true : https://stackoverflow.com/questions/13808020/include-an-svg-hosted-on-github-in-markdown

**Comment by jreidinger** [comment]
I add note and merge it for now, as improvement can be done as another pull request.

**Pull request merged**